frontend/packages/dev-console/integration-tests/support/step-definitions/build-config/edit-build-config.ts

```diff
@@ -0,0 +1,216 @@
+  (imageStream: string, imageStreamTag: string) => {
+    cy.get(buildConfigPO.buildFrom.imageStreamDropdown).click();
+    cy.byLegacyTestID('dropdown-text-filter').type(imageStream);
+    cy.byTestID('dropdown-menu-item-link')
+      .eq(0)
+      .click();
+    cy.get(buildConfigPO.buildFrom.imageStreamTagDropdown).click();
+    cy.byLegacyTestID('dropdown-text-filter').type(imageStreamTag);
+    cy.byTestID('dropdown-menu-item-link')
+      .eq(1)
+      .click();
+  },
+);
+
+When(
+  'user enters Name and Value as {string} and {string} respectively in Environment Variables',
+  (name: string, value: string) => {
+    cy.byTestID('pairs-list-name')
+      .scrollIntoView()
+      .type(name);
+    cy.byTestID('pairs-list-value')
+      .scrollIntoView()
+      .type(value);
+  },
+);
+
+When('user selects External container image option from Build from dropdown', () => {
+  cy.get(buildConfigPO.buildFrom.buildTypeDropdown).click();
+  cy.byTestDropDownMenu('dockerImage').click();
+});
+
+When('user enters image registry as {string}', (containerImage: string) => {
+  cy.get(buildConfigPO.imageRegistryField)
+    .clear()
+    .type(containerImage);
+});
+
+When('user click Save button on Edit build Config page', () => {
+  cy.byLegacyTestID('submit-button').click();
+});
+
+Then(
+  'user will see Git Repository as {string} and Build from as {string} imagestream',
+  (gitRepoUrl: string, imageStream: string) => {
+    cy.byTestSelector('details-item-value__Git repository').should('have.text', gitRepoUrl);
+    cy.byTestSelector('details-item-value__Build from')
+      .byTestID(imageStream)
+      .should('have.text', imageStream);
+  },
+);
+
+Given('user has applied the yaml {string}', (yamlFile: string) => {
+  const yamlFileName = `testData/YAMLs/BuildConfig/${yamlFile}`;
+  navigateTo(devNavigationMenu.Add);
+  cy.byTestID('item import-yaml').click();
+  yamlEditor.isLoaded();
+  yamlEditor.setEditorContent(yamlFileName);
+  cy.byTestID('save-changes').click();
+});
+
+When('user clicks on action menu of build config', () => {
+  cy.byLegacyTestID('actions-menu-button').click();
+  cy.byLegacyTestID('action-items').should('be.visible');
+});
+
+When('user selects the option Edit BuildConfig', () => {
+  cy.byTestActionID('Edit BuildConfig').click();
+});
+
+When('user clicks Show advanced Git options', () => {
+  cy.contains('Show advanced Git options').click();
+  cy.byTestID('advanced-git-options').should('be.visible');
+});
+
+When('user changes value of Context Dir to {string}', (dir: string) => {
+  cy.get(buildConfigPO.contentDirectoryField)
+    .clear()
+    .type(dir);
+});
+
+Then('user will see Context dir as {string}', (dir: string) => {
+  cy.byTestSelector('details-item-value__Context dir').should('have.text', dir);
+});
+
+When('user click Save button on a Edit build Config page', () => {
+  cy.byLegacyTestID('submit-button').click();
```

Comment:
Do check for already existing id's
```suggestion
  cy.get(formPO.create).click();
```